node/request.js

```diff
@@ -224,19 +224,36 @@ TChannelRequest.prototype.resend = function resend() {
     var perAttemptStart = self.timers.now();
 
     var conn = peer.connect();
-    if (!peer.isConnected() && conn.handler) {
-        conn.on('identified', onIdentified);
-        var timer = self.timers.setTimeout(onIdentifyTimeout, self.timeout - self.elapsed);
-        return;
+
+    if (conn.closing) {
+        onConnectionError(conn.closeError);
     } else {
-        onIdentified();
+        conn.closeEvent.on(onConnectionError);
+
+        if (!peer.isConnected() && conn.handler) {
+            conn.identifiedEvent.on(onIdentified);
+            var timer = self.timers.setTimeout(onIdentifyTimeout, self.timeout - self.elapsed);
+            return;
+        } else {
+            onIdentified();
+        }
+    }
+
+    function onConnectionError(err) {
+        conn.closeEvent.removeListener(onConnectionError);
+        if (timer) {
+            self.timers.clearTimeout(timer);
+        }
+        self.emitError(err);
```

Comment:
`err || new Error('connection reset')`

We dont know if `err` will always exist or not.